packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/arrow.dart

```diff
@@ -0,0 +1,280 @@
+  }
+}
+
+class _LeftwardsArrowPaintStrategy implements _ArrowPaintStrategy {
+  @override
+  void paint({Canvas canvas, Size size, Paint paint, double headSize}) {
+    final arrowHeadSizeDividedByTwo = headSize / 2;
+    final p1 = Offset.zero;
+    final p2 = Offset(headSize, -arrowHeadSizeDividedByTwo);
+    final p3 = Offset(headSize, arrowHeadSizeDividedByTwo);
+    canvas.drawPath(_ArrowPaintStrategy.pathForArrowHead(p1, p2, p3), paint);
+    final lineStartingPoint = Offset(headSize, 0);
+    final lineEndingPoint = Offset(size.width, 0);
+    canvas.drawLine(lineStartingPoint, lineEndingPoint, paint);
+  }
+}
+
+class _DownwardsArrowPaintStrategy implements _ArrowPaintStrategy {
+  @override
+  void paint({Canvas canvas, Size size, Paint paint, double headSize}) {
+    final arrowHeadSizeDividedByTwo = headSize / 2;
+    final arrowHeadStartingY = size.height - headSize;
+    final lineStartingPoint = Offset.zero;
+    final lineEndingPoint = Offset(0, arrowHeadStartingY);
+    canvas.drawLine(lineStartingPoint, lineEndingPoint, paint);
+    final p1 = Offset(0, size.height);
+    final p2 = Offset(-arrowHeadSizeDividedByTwo, arrowHeadStartingY);
+    final p3 = Offset(arrowHeadSizeDividedByTwo, arrowHeadStartingY);
+    canvas.drawPath(_ArrowPaintStrategy.pathForArrowHead(p1, p2, p3), paint);
+  }
+}
+
+class _RightwardsArrowPaintStrategy implements _ArrowPaintStrategy {
+  @override
+  void paint({Canvas canvas, Size size, Paint paint, double headSize}) {
+    final arrowHeadSizeDividedByTwo = headSize / 2;
+    final arrowHeadStartingX = size.width - headSize;
+    final lineStartingPoint = Offset.zero;
+    final lineEndingPoint = Offset(arrowHeadStartingX, 0);
+    canvas.drawLine(lineStartingPoint, lineEndingPoint, paint);
+    final p1 = Offset(size.width, 0);
+    final p2 = Offset(arrowHeadStartingX, -arrowHeadSizeDividedByTwo);
+    final p3 = Offset(arrowHeadStartingX, arrowHeadSizeDividedByTwo);
+    canvas.drawPath(_ArrowPaintStrategy.pathForArrowHead(p1, p2, p3), paint);
+  }
+}
+
+class _ArrowPainter extends CustomPainter {
+  _ArrowPainter({
+    this.arrowHeadSize = 15.0,
+    this.strokeWidth = 2.0,
+    this.color = Colors.white,
+    @required this.strategy,
+  })  : assert(color != null),
+        assert(arrowHeadSize != null),
+        assert(strokeWidth != null),
+        assert(strategy != null);
+
+  final double arrowHeadSize;
+  final double strokeWidth;
+  final Color color;
+  final _ArrowPaintStrategy strategy;
+
+  @override
+  bool shouldRepaint(CustomPainter oldDelegate) =>
+      !(oldDelegate is _ArrowPainter &&
+          arrowHeadSize == oldDelegate.arrowHeadSize &&
+          strokeWidth == oldDelegate.strokeWidth &&
+          color == oldDelegate.color &&
+          strategy == oldDelegate.strategy);
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final paint = Paint()
+      ..color = color
+      ..strokeWidth = strokeWidth;
+
+    strategy.paint(
```

Comment:
I think we can probably remove the strategy and instead add slightly more generic arrow drawing code directly here that handles all arrow directions.